src/fuzz_introspector/frontends/src/frontend_go.py

```diff
@@ -0,0 +1,610 @@
+
+        if function_list:
+            report['All functions'] = {}
+            report['All functions']['Elements'] = function_list
+
+        with open(report_name, 'w', encoding='utf-8') as f:
+            f.write(yaml.dump(report))
+
+    def get_source_codes_with_harnesses(self) -> list[SourceCodeFile]:
+        """Gets the source codes that holds libfuzzer harnesses."""
+        harnesses = []
+        for source_code in self.source_code_files:
+            if source_code.has_libfuzzer_harness():
+                harnesses.append(source_code)
+        return harnesses
+
+    def extract_calltree(self,
+                         source_file: str,
+                         source_code: Optional[SourceCodeFile] = None,
+                         function: str = None,
+                         visited_functions: set[str] = None,
+                         depth: int = 0,
+                         line_number: int = -1) -> str:
+        """Extracts calltree string of a calltree so that FI core can use it."""
+        if not visited_functions:
+            visited_functions = set()
+
+        if not function:
+            function = source_code.get_entry_function_name()
+
+        line_to_print = '  ' * depth
+        line_to_print += function
+        line_to_print += ' '
+        line_to_print += source_file
+
+        if not source_code:
+            source_code = self.find_source_with_func_def(function)
+
+        line_to_print += ' '
+        line_to_print += str(line_number)
+
+        line_to_print += '\n'
+        if not source_code:
+            return line_to_print
+
+        func = source_code.get_function_node(function)
+        if not func:
+            return line_to_print
+
+        callsites = func.base_callsites()
+
+        if function in visited_functions:
+            return line_to_print
+
+        visited_functions.add(function)
+        for cs, line_number in callsites:
+            line_to_print += self.extract_calltree(
+                source_code.source_file,
+                function=cs,
+                visited_functions=visited_functions,
+                depth=depth + 1,
+                line_number=line_number)
+        return line_to_print
+
+    def find_source_with_func_def(
+            self, target_function_name: str) -> Optional[SourceCodeFile]:
+        """Finds the source code with a given function."""
+        for source_code in self.source_code_files:
+            if source_code.has_function_definition(target_function_name):
+                return source_code
+
+        return None
+
+
+class FunctionMethod():
+    """Wrapper for a General Declaration for function/method"""
+
+    def __init__(self, root: Node, tree_sitter_lang: Language,
+                 source_code: SourceCodeFile, is_function: bool):
+        self.root = root
+        self.tree_sitter_lang = tree_sitter_lang
+        self.parent_source = source_code
+        self.is_function = is_function
```

Comment:
what is `is_function` used for?

Reply:
it is used to distinguish functions from methods. But currently it is just a unused flag. I am storing this in order for future reference if some process needed to identify them.